TESTING.asciidoc

```diff
@@ -438,84 +433,51 @@ that'd consume a ton of ram.
 
-Running the packaging tests through gradle can take a while because it will start
-and stop the VM each time. You can iterate faster by keeping the VM up and running
-the tests directly.
-
-The packaging tests use a random seed to determine which past version to use for
-testing upgrades. To use a single past version fix the test seed when running
-the commands below (see <<Seed and repetitions.>>)
-
-First build the packaging tests and their dependencies
-
---------------------------------------------
-./gradlew :qa:vagrant:setupPackagingTest
---------------------------------------------
-
-Then choose the VM you want to test on and bring it up. For example, to bring
-up Debian 9 use the gradle command below. Bringing the box up with vagrant directly
-may not mount the packaging test project in the right place. Once the VM is up, ssh
-into it
-
---------------------------------------------
-./gradlew :qa:vagrant:vagrantDebian9#up
-vagrant ssh debian-9
---------------------------------------------
-
-Now inside the VM, start the packaging tests from the terminal. There are two packaging
-test projects. The old ones are written with https://github.com/sstephenson/bats[bats]
-and only run on linux. To run them do
-
---------------------------------------------
-cd $PACKAGING_ARCHIVES
-
-# runs all bats tests
-sudo bats $BATS_TESTS/*.bats
-
-# you can also pass specific test files
-sudo bats $BATS_TESTS/20_tar_package.bats $BATS_TESTS/25_tar_plugins.bats
---------------------------------------------
+Because our packaging tests are capable of testing many combinations of OS (Windows
+or Linux), package type (zip file or RPM), Elasticsearch distribution type (default
+or OSS), and so forth, it's faster to develop against smaller subsets of the tests.
+For example, to run tests for the default archive distribution on Fedora 28:
 
-The new packaging tests are written in Java and run on both linux and windows. On
-linux (again, inside the VM)
+-----------------------------------------------------------
+./gradlew :qa:os:fedora-28:distroTest.default-linux-archive
+-----------------------------------------------------------
 
---------------------------------------------
-# run the full suite
-sudo bash $PACKAGING_TESTS/run-tests.sh
+These test tasks can use the `--tests`, `--info`, and `--debug` parameters just like
+non-OS tests can. For example:
 
-# run specific test cases
-sudo bash $PACKAGING_TESTS/run-tests.sh \
-  org.elasticsearch.packaging.test.DefaultWindowsZipTests \
-  org.elasticsearch.packaging.test.OssWindowsZipTests
---------------------------------------------
+-----------------------------------------------------------
+./gradlew :qa:os:fedora-28:distroTest.default-linux-archive \
+  --tests "com.elasticsearch.packaging.test.ArchiveTests"
+-----------------------------------------------------------
 
-or on Windows, from a terminal running as Administrator
+If you need to work on the packaging test framework or run Gradle tests from
```

Comment:
Yeah, this bit is pretty goofy. If this is something we expect folks to do, we should bake this capability into the plugin and provide a task for folks just to spin up a "ready-to-go" VM.

Reply:
This should be possible, but we should give sufficient large warnings/notification that the task is going to leave the VM running, and other guards (eg nothing should depend on such a task, it should only be run directly). A followup issue for discussion/impl details seems fine.

Reply:
Cool — I'll remove this section but make a new issue for team discussion. Thanks.

Reply:
I've written a follow-up ticket for discussion: https://github.com/elastic/elasticsearch/issues/48087